clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -5408,7 +5408,10 @@ def note_dependent_function_template_spec_discard_reason : Note<
   "not a member of the enclosing %select{class template|"
   "namespace; did you mean to explicitly qualify the specialization?}1}0">;
 def warn_invalid_specialization : Warning<
-  "%0 should not be specialized">,
+  "%0 cannot be specialized">,
```

Comment:
```suggestion
  "%0 cannot be specialized%select{|: %2}1">,
```

And then change the diagnostics to `Diag(Loc, diag::warn_invalid_specialization) << VarOrClassTemplate << !Message.empty() << Message;`